connect-inject/container_init.go

```diff
@@ -111,14 +112,25 @@ func (h *Handler) containerInit(pod *corev1.Pod) (corev1.Container, error) {
 		for _, raw := range strings.Split(raw, ",") {
 			parts := strings.SplitN(raw, ":", 3)
 
-			var datacenter, service_name, prepared_query string
+			var datacenter, service_name, prepared_query, namespace string
 			var port int32
 			if parts[0] == "prepared_query" {
 				port, _ = portValue(pod, strings.TrimSpace(parts[2]))
 				prepared_query = strings.TrimSpace(parts[1])
 			} else {
 				port, _ = portValue(pod, strings.TrimSpace(parts[1]))
-				service_name = strings.TrimSpace(parts[0])
+
+				// Parse the namespace if provided
+				if data.ConsulNamespace != "" {
+					pieces := strings.SplitN(parts[0], ".", 2)
```

Comment:
If somebody provides an invalid string `service.namespace.dc`, then the result in `pieces` will be `[service namespace.dc]`. We should probably detect it and either log it and ignore everything past the second dot, or error.